File: TESTDB/src/Data/Toilet.java
package Data;


import javax.persistence.*;
import java.util.HashMap;
import java.util.LinkedHashMap;
import java.util.Map;

@Entity
public class Toilet implements Care.CareInformation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private boolean flush = false;

    @Transient
    private LinkedHashMap<String, Boolean> map = new LinkedHashMap<>();

    Toilet() {
    }

    Toilet NewUnicorn() {
        map.put("flush", false);
        return this;
    }

    @Override
    public LinkedHashMap<String, Boolean> getCategories() {
        if (!isFlushed()) {
            map.put("flush", isFlushed());
        }
        return map;
    }

    @Override
    public String toString() {
        return getClass().getSimpleName();
    }

    @Override
    public String save(String key) {
        if (key.equals("flush")) {
            setFlushed(true);
            map.remove(key);
        }
        return "is glad that you helped him " + key;
    }


    public int getId() {
        return id;
    }

    private void setId(int id) {
        this.id = id;
    }


    public boolean isFlushed() {
        return flush;
    }

    private void setFlushed(boolean flushed) {
        this.flush = flushed;
    }

}

File: TESTDB/src/Dao/Toilet.kt
package Dao


import javax.persistence.*
import java.util.LinkedHashMap

@Entity
class Toilet : Care.CareInformation, DaoMaster() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int = 0
        private set

    var flush = false

    @Transient
    private var unicornProperties = LinkedHashMap<String, Boolean>()


    override fun newUnicorn(): Toilet {
        unicornProperties = setNewUnicorn(unicornProperties)
        return this
    }

    override fun checkForUpdates() {
        unicornProperties = checkAndReturnUpdates(unicornProperties)

    }

    override fun save(key: String): String {
        unicornProperties.remove(saveChanges(key))
        return "is glad that you helped him $key"
    }

    override fun getCategories(): LinkedHashMap<String, Boolean> {
        return unicornProperties
    }

    override fun toString(): String {
        return javaClass.simpleName
    }
}
